integration/default_apps_test.go

```diff
@@ -30,70 +28,73 @@ func testDefaultApps(t *testing.T, context spec.G, it spec.S) {
-			Expect(docker.Image.Remove.Execute(image.ID)).To(Succeed())
+			for id := range containers {
+				Expect(docker.Container.Remove.Execute(id)).To(Succeed())
+			}
+			for id := range images {
+				Expect(docker.Image.Remove.Execute(id)).To(Succeed())
+			}
 			Expect(docker.Volume.Remove.Execute(occam.CacheVolumeNames(name))).To(Succeed())
 			Expect(os.RemoveAll(source)).To(Succeed())
 		})
 
 		context("given a source application with .NET Core 6", func() {
-			it("should build a working OCI image", func() {
+			it("should build (and rebuild) a working OCI image", func() {
 				var err error
 				source, err := occam.Source(filepath.Join("testdata", "source_6_app"))
 				Expect(err).NotTo(HaveOccurred())
 
-				var logs fmt.Stringer
-				image, logs, err = pack.WithNoColor().Build.
-					WithBuildpacks(
-						icuBuildpack,
-						dotnetCoreRuntimeBuildpack,
-						dotnetCoreAspNetBuildpack,
-						dotnetCoreSDKBuildpack,
-						buildpack,
-						dotnetExecuteBuildpack,
-					).
-					WithEnv(map[string]string{
-						"BP_DOTNET_PUBLISH_FLAGS": "--verbosity=normal",
-					}).
-					Execute(name, source)
-				Expect(err).NotTo(HaveOccurred(), logs.String())
-
-				Expect(logs).To(ContainLines(
-					MatchRegexp(`    Running 'dotnet publish .* --verbosity=normal'`),
-				))
-
-				container, err = docker.Container.Run.
-					WithEnv(map[string]string{"PORT": "8080"}).
-					WithPublish("8080").
-					WithPublishAll().
-					Execute(image.ID)
-				Expect(err).NotTo(HaveOccurred())
-
-				Eventually(container).Should(BeAvailable())
-
-				response, err := http.Get(fmt.Sprintf("http://localhost:%s", container.HostPort("8080")))
-				Expect(err).NotTo(HaveOccurred())
-				defer response.Body.Close()
-
-				Expect(response.StatusCode).To(Equal(http.StatusOK))
-
-				content, err := io.ReadAll(response.Body)
-				Expect(err).NotTo(HaveOccurred())
-				Expect(string(content)).To(ContainSubstring("source_6_app"))
+				for i := 0; i < 2; i++ {
+					var logs fmt.Stringer
+					image, logs, err = pack.WithNoColor().Build.
+						WithBuildpacks(
+							icuBuildpack,
+							dotnetCoreRuntimeBuildpack,
+							dotnetCoreAspNetBuildpack,
+							dotnetCoreSDKBuildpack,
+							buildpack,
+							dotnetExecuteBuildpack,
+						).
+						WithEnv(map[string]string{
+							"BP_DOTNET_PUBLISH_FLAGS": "--verbosity=normal",
+						}).
+						Execute(name, source)
+					Expect(err).NotTo(HaveOccurred(), logs.String())
+					images[image.ID] = ""
+
+					Expect(logs).To(ContainLines(
+						MatchRegexp(`    Running 'dotnet publish .* --verbosity=normal'`),
+					))
+
+					container, err = docker.Container.Run.
+						WithEnv(map[string]string{"PORT": "8080"}).
+						WithPublish("8080").
+						WithPublishAll().
+						Execute(image.ID)
+					Expect(err).NotTo(HaveOccurred())
+					containers[container.ID] = ""
+
+					Eventually(container).Should(BeAvailable())
+					Eventually(container).Should(Serve(ContainSubstring("source_6_app")).OnPort(8080))
+				}
```

Comment:
@fg-j it's not super clear to me from this test how to tell that the cache is being reused on the rebuild. Am I missing something?

Reply:
So the idea here is to test that loading the cache when it's non-empty and running `dotnet publish` works as expected. Since the [intermediate build layer is marked `cache`](https://github.com/paketo-buildpacks/dotnet-publish/blob/8446e5abb4bb608cc444fb92370fd197e2a7c7e1/build.go#L107) and we [always try to load the cache before building](https://github.com/paketo-buildpacks/dotnet-publish/blob/8446e5abb4bb608cc444fb92370fd197e2a7c7e1/dotnet_publish_process.go#L35), building and then rebuilding tests that a) the cache is saved correctly after the first build and b) the cache is useable during the second build. Analogous to [this rebuild test](https://github.com/paketo-buildpacks/go-build/blob/2a28b61aa1363447225c3bbae8b9fad1a1bed1b5/integration/rebuild_test.go#L55) in the go-build buildpack.

Reply:
Ah ok, the success of the build is enough to know that the cache is used by nature of the code.

I think I was expecting to see an assertion around the cache layer SHA, but I realize we can't do that because it's only a `cache` layer, not `launch`.